src/TorchSharp/NN/Module.cs

```diff
@@ -129,6 +139,54 @@ public Module to(DeviceType deviceType, int deviceIndex = -1)
                         THSNN_Module_to_device(handle, (int)deviceType, deviceIndex);
                         torch.CheckForErrors();
 
+                        foreach (var (_, sm) in named_children()) sm.to(deviceType, deviceIndex);
+
+                        foreach (var field in this.GetType().GetFields(BindingFlags.NonPublic | BindingFlags.Public | BindingFlags.Instance).Where(f => !f.Name.Contains("__BackingField"))) {
```

Comment:
This is very uncomfortable, and will perhaps only work for C#.  I need to understand better why we'd be resorting to such things.

Reply:
So whatever is going on here needs to be documented properly, that is, what does it actually mean for the programming model for Modules?

Reply:
Thanks!

Specifically, what are you uncomfortable with? The "BackingField" string comparison filter, or something else?

Reply:
The logic was introduced when I added properties as valid holders of parameters and buffers. In order to avoid double-counting, I have to filter out the backing field for such properties. Now that you bring it up, I realize that it may not work the same in F#. The F# examples run to completion, but it may be double-counting (wasting time).

Reply:
I guess we could avoid this mess by only allowing buffers and parameters to be held in properties, not in fields.

Reply:
@dsyme -- I pulled the support for properties holding parameters and buffers out of this PR. We can resurface it later, with proper cross-language support.